test/jdk/java/io/SequenceInputStream/TransferTo.java

```diff
@@ -126,6 +128,49 @@ public void testStreamContents() throws Exception {
                 outputStreamProvider, createRandomBytes(4096, 0), 0, 4096);
     }
 
+    /*
+     * Special case: Assert subsequent input stream is read when preceding stream already was MAX_VALUE long.
+     * Note: Not testing actual content as it requires multiple GBs of memory and long time.
+     */
+    @Test
+    public void testHugeStream() throws Exception {
+        InputStream is1 = repeat(0, Long.MAX_VALUE);
+        InputStream is2 = repeat(0, 1);
+        assertNotEquals(is1.available(), 0);
+        assertNotEquals(is2.available(), 0);
+        SequenceInputStream sis = new SequenceInputStream(is1, is2);
+        OutputStream nos = OutputStream.nullOutputStream();
+        sis.transferTo(nos);
+        assertEquals(is1.available(), 0);
+        assertEquals(is2.available(), 0);
+    }
+
+    /*
+     * Produces an input stream that returns b count times.
+     * Builds a dysfunctional mock that solely implements
+     * available() and transferTo() particually,
+     * but fails with any other operation.
+     */
+    private static InputStream repeat(int b, long count) {
```

Comment:
`b` parameter seems to be unused. Wdyt of removing it?

Reply:
Kept it for future use: If in future we like to add `write(b)` in `transferTo` then we do not have to rework the call sites.